src/ComplexTypes/BaseLocation.php

```diff
@@ -0,0 +1,134 @@
+<?php namespace DivideBV\Postnl\ComplexTypes;
+
+class BaseLocation extends BaseType
+{
+
+    /**
+     * @var string $AllowSundaySorting
+     */
+    protected $AllowSundaySorting = null;
+
+    /**
+     * @var string $DeliveryDate
+     */
+    protected $DeliveryDate = null;
+
+    /**
+     * @var string[] $DeliveryOptions
+     */
+    protected $DeliveryOptions = null;
+
+    /**
+     * @var string $OpeningTime
+     */
+    protected $OpeningTime = null;
+
+    /**
+     * @var string[] $Options
+     */
+    protected $Options = null;
+
+    /**
+     * @param string $AllowSundaySorting
+     * @param string $DeliveryDate
+     * @param array $DeliveryOptions
+     * @param array $Options
+     */
+    public function __construct($AllowSundaySorting, $DeliveryDate, $DeliveryOptions, $Options)
+    {
+        $this->setAllowSundaySorting($AllowSundaySorting);
+        $this->setDeliveryDate($DeliveryDate ?: (new \DateTime('next monday'))->format('d-m-Y'));
+        $this->setDeliveryOptions($DeliveryOptions);
+        $this->setOptions($Options);
+    }
+
+    /**
+     * @return string
+     */
+    public function getAllowSundaySorting()
+    {
+        return $this->AllowSundaySorting;
+    }
+
+    /**
+     * @param string $AllowSundaySorting
+     * @return BaseLocation
```

Comment:
Strictly speaking, this is not correct. It returns `$this`, which can be a class that extends `BaseLocation` (of which there are 2 in the current implementation). It would probably be better to do `@return $this` (see https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#appendix-a-types).

This theoretically goes for all ComplexTypes really, though for the already existing ones there's no inheritance of any `setFoo` methods.